### ISSUE TITLE
Job Queuing & Service Workers

### ISSUE THREAD

**Issue by kisuka** [SmartCasual/jingle-jam#77]
Not super fluent in Ruby development, but I was poking around the codebase and was trying to determine if the stack is setup for queued messages with service workers consuming the queue to perform large operations in a FIFO flow. The fact that this service historically is always under large degrees of traffic, I'd imagine that message queuing would be an extremely important must for scaling purposes. The most I saw was that Jobs directory, but it didn't seem fleshed out yet, unless I was missing something. Queuing would be important for many operations such as sending emails, code unlocking, and donation processing. Is everything planned to be a monolith or would workers be split out so they could scale independent from the web services? Also would the message queue be offloaded to it's own service like SQS or Beanstalkd? Or is it all pulling from the DB?

**Comment by elliotcm** [comment]
Hi @kisuka.

Rails has a sub-framework called ActiveJob, and the classes in the `jobs` folder inherit from that.  They are background workers, in this case backed by Sidekiq.  These cover key assignments, currently.

Emails are sent via another sub-framework called ActionMailer, which is also integrated into ActiveJob, so we can make any email sending async by calling `deliver_later` instead of `deliver_now`.

The only other part I'd like to make async is the donation processing itself (after receiving webhooks from Stripe), and that's something I'm currently working on.

So, long story short, backend jobs are (or will be) in use where needed.
